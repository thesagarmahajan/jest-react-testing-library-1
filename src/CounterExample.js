import { useState } from "react"

export default function CounterExample(){
    let [count, setCount] = useState(0)
    return (<>
        <h1>{count}</h1>
        <button onClick={()=>setCount(count+1)}>+</button>
        <button onClick={()=>count!=0?setCount(count-1):setCount(0)}>-</button>
        <button onClick={()=>setCount(0)}>Reset</button>
    </>)
}