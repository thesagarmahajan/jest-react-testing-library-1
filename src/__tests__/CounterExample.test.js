import { render } from "@testing-library/react";
import CounterExample from "../CounterExample";
import { screen } from "@testing-library/react";
import user from '@testing-library/user-event'
 
test.only('Testing CounterExample', () => {
    render(<CounterExample />)

    let [incButton, decButton, resetButton] = screen.getAllByRole('button')
    let heading = screen.getByRole('heading')
    
    expect(heading.textContent).toBe("0")

    user.click(incButton)
    user.click(incButton)
    expect(heading.textContent).toBe("2")
    user.click(decButton)
    user.click(decButton)
    expect(heading.textContent).toBe("0")
    user.click(decButton)
    expect(heading.textContent).not.toBe("-1")
    user.click(resetButton)
    expect(heading.textContent).toBe("0")
});