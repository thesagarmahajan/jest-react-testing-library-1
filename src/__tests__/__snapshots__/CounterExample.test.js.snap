// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demonstrating Snapshot Testing 1`] = `
Array [
  <h1>
    Count: 
    0
  </h1>,
  <button
    onClick={[Function]}
  >
    +
  </button>,
  <button
    onClick={[Function]}
  >
    -
  </button>,
  <button
    onClick={[Function]}
  >
    Reset
  </button>,
]
`;
